scripts/import/import_EPMC.pl

```diff
@@ -993,9 +1005,253 @@ sub get_current_citations{
+
+    # Get publication with same PMID from study table
+    my $publication = $pub_ad->fetch_by_pmid($external_reference);
+
+    next unless !defined($publication);
+
+    # Get attrib id for source - some are null 
+    my $source_attrib_id;
+    if(defined $study_type){
+      $source_attrib_id = $citation_attribs->{$study_type};
+      die "No attrib of type 'citation_source' was found for '$study_type'!\n" unless defined $source_attrib_id;
+    }
+    else{
+      # Get source name from source table (dbGaP)
+      my $source_obj = $source_ad->fetch_by_dbID($source_id);
+      my $source_name = $source_obj->name();
+      $source_attrib_id = $citation_attribs->{$source_name};
+      die "No attrib of type 'citation_source' was found for '$source_name'!\n" unless defined $source_attrib_id;
+    }
+
+    # Get publication that is not in publication table 
+    my $pub_data = get_epmc_data("search?query=ext_id:$external_reference%20src:med");
+
+    if(defined $pub_data->{resultList}->{result}->{title}){
+      my $pub_title = $pub_data->{resultList}->{result}->{title};
+      my $pub_pmcid = $pub_data->{resultList}->{result}->{pmcid};
+      my $pub_author = $pub_data->{resultList}->{result}->{authorString};
+      my $pub_year = $pub_data->{resultList}->{result}->{pubYear};
+      my $pub_doi = $pub_data->{resultList}->{result}->{doi};
+      
+      my @variant_list;
+      $rsid_sth->execute($study_id);
+      my $rsids = $rsid_sth->fetchall_arrayref();
+      
+      foreach my $rsid (@{$rsids}){
+        my $v = $var_ad->fetch_by_name($rsid->[0]);
+        if (defined $v){
+          push @variant_list, $v;
+        }
+      }
+
+      ### create new object
+      my $publication = Bio::EnsEMBL::Variation::Publication->new( 
+                -title    => $pub_title,
+                -authors  => $pub_author,
+                -pmid     => $external_reference,
+                -pmcid    => $pub_pmcid || undef,
+                -year     => $pub_year,
+                -doi      => $pub_doi,
+                -ucsc_id  => undef,
+                -variants => \@variant_list,
+                -adaptor  => $pub_ad
+                );
+
+       $pub_ad->store($publication,$source_attrib_id);
+    }
+  }
+}
+
+sub process_phenotype_feature_attrib {
+  my $dba = shift;
+  my $var_ad = shift;
+  my $pub_ad = shift;
+  my $source_ad = shift;
+  my $citation_attribs = shift;
+
+  my $attrib_type_sth = $dba->dbc()->prepare(qq[ select attrib_type_id
+                                                 from attrib_type
+                                                 where code = 'pubmed_id' ]);
+
+  $attrib_type_sth->execute()||die;
+  my $attrib_type_id = $attrib_type_sth->fetchall_arrayref();
+
+  die "No attribute type 'pubmed_id' found\n" unless defined $attrib_type_id->[0]->[0];
+
+  my $attrib_type_id_v = $attrib_type_id->[0]->[0];
+
+  my $pheno_feature_sth = $dba->dbc()->prepare(qq[ select phenotype_feature_id, value 
```

Comment:
joining to phenotype_feature and pulling out the variant names & source in one statement should be quicker here